src/components/Proposal/View/Budget/CompetingProposalsSidebar.tsx

```diff
@@ -0,0 +1,248 @@
+import React, { useEffect, useMemo, useState } from 'react'
+
+import useFormatMessage from 'decentraland-gatsby/dist/hooks/useFormatMessage'
+import TokenList from 'decentraland-gatsby/dist/utils/dom/TokenList'
+import { Close } from 'decentraland-ui/dist/components/Close/Close'
+import snakeCase from 'lodash/snakeCase'
+import Sidebar from 'semantic-ui-react/dist/commonjs/modules/Sidebar/Sidebar'
+
+import { BudgetWithContestants, CategoryBudgetWithContestants } from '../../../../entities/Budget/types'
+import { ProposalAttributes } from '../../../../entities/Proposal/types'
+import { toNewGrantCategory } from '../../../../entities/QuarterCategoryBudget/utils'
+import { getFormattedPercentage } from '../../../../helpers'
+import DistributionBar from '../../../Common/DistributionBar/DistributionBar'
+import { DistributionBarItemProps } from '../../../Common/DistributionBar/DistributionBarItem'
+import { GrantRequestSectionCard } from '../../../GrantRequest/GrantRequestSectionCard'
+
+import CompetingProposal from './CompetingProposal'
+import './CompetingProposalsSidebar.css'
+
+/* eslint-disable @typescript-eslint/ban-types */
+function getContestingProposalsItems(
+  t: <V extends {}>(id?: string | null, values?: V | undefined) => string,
+  proposal: ProposalAttributes,
+  categoryBudget: CategoryBudgetWithContestants,
+  highlightedContestant: string | null,
+  setHighlightedContestant: (value: ((prevState: string | null) => string | null) | string | null) => void
+) {
+  const items: DistributionBarItemProps[] = []
+  categoryBudget?.contestants.forEach((contestant) => {
+    if (contestant.id !== proposal.id) {
+      items.push({
+        value: contestant.size,
+        className: 'ContestedBudgetBar',
+        selected: highlightedContestant === contestant.id,
+        onHover: (e: React.MouseEvent<unknown>) => {
+          setHighlightedContestant(contestant.id)
+        },
+        onBlur: () => {
+          setHighlightedContestant(null)
+        },
+        popupContent: {
+          title: contestant.title,
+          content: <span>{`$${t('general.number', { value: contestant.size })}`}</span>,
```

Comment:
Perhaps it would be good to use the currency formatter in this case for consistency with the rest of the code

You can take as an example `BudgetBanner.tsx`